src/jit/hwintrinsiclistarm64.h

```diff
@@ -0,0 +1,118 @@
+//                 ISA              Function name                            ival        SIMD size       NumArg                                                                                                     instructions                                                                                                     Category                            Flags
+//                                                                                                                   {TYP_BYTE,              TYP_UBYTE,          TYP_SHORT,          TYP_USHORT,         TYP_INT,            TYP_UINT,           TYP_LONG,           TYP_ULONG,          TYP_FLOAT,          TYP_DOUBLE}
+// ***************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
+//  AdvSimd Intrinsics
+HARDWARE_INTRINSIC(AdvSimd,         Abs,                                       -1,              -1,           1,     {INS_invalid,           INS_abs,            INS_invalid,        INS_abs,            INS_invalid,        INS_abs,            INS_invalid,        INS_invalid,        INS_fabs,           INS_invalid},           HW_Category_SimpleSIMD,             HW_Flag_NoContainment|HW_Flag_UnfixedSIMDSize)
+HARDWARE_INTRINSIC(AdvSimd,         AbsScalar,                                 -1,               8,           1,     {INS_invalid,           INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_fabs,           INS_fabs},              HW_Category_SIMDScalar,             HW_Flag_NoContainment)
+HARDWARE_INTRINSIC(AdvSimd,         Add,                                       -1,              -1,           2,     {INS_add,               INS_add,            INS_add,            INS_add,            INS_add,            INS_add,            INS_add,            INS_add,            INS_fadd,           INS_invalid},           HW_Category_SimpleSIMD,             HW_Flag_NoContainment|HW_Flag_Commutative|HW_Flag_UnfixedSIMDSize)
+HARDWARE_INTRINSIC(AdvSimd,         AddScalar,                                 -1,               8,           2,     {INS_add,               INS_add,            INS_add,            INS_add,            INS_add,            INS_add,            INS_add,            INS_add,            INS_fadd,           INS_fadd},              HW_Category_SIMDScalar,             HW_Flag_NoContainment|HW_Flag_Commutative)
```

Comment:
@TamarChristinaArm.

What is the semantics around the unused bits for a given vector register?

That is, on Arm64 the vector register is 128-bits in length. However, some encodings only access the first 64-bits (`Vector64<T>` operations) and some instructions only access the first element (`scalar` instructions only access the first `T`).

So, for say `AddScalar` which takes in `Vector64<float> left, Vector64<float> right` and returns a `Vector64<T>`; the contents of `result[0] = left[0] + right[0]`, but what is the contents of `result[1]` and what is the result of indice 3 and 4 of the backing register (bits 64-96 and bits 96-128)?

Is it cleared to zero, is it preserved to the last value in the register, etc?

The same question goes for `Vector64<T> Add(Vector64<T> left, Vector64<T> right)` (what is the contents of bits 64-96 and bits 96-128 of the backing register)?

Reply:
I'm asking because I'm seeing for:
```
 left: (0.41178197, 0.0033083581)
right: (0.11088856, 0.89114616)
```

You get:
```
result: (0.5226705, 0)
```

and I'm wanting to find out if this is deterministically `0` or if it is the last value contained in those bits.

Reply:
The default is unless otherwise specified by the instruction (such is `inserts`, or `high` operations) the unused bits of a register is always cleared to `0`.

Reply:
> So, for say AddScalar which takes in Vector64<float> left, Vector64<float> right and returns a Vector64<T>; the contents of result[0] = left[0] + right[0], but what is the contents of result[1] and what is the result of indice 3 and 4 of the backing register (bits 64-96 and bits 96-128)?

They're all cleared to `0` in this case. They wouldn't be for an insert (e.g. `INS`) a structure load into one lane (e.g. `LD3...[<lane>]`) or on some of the instructions which have a `second part` (those generally end with a `2` in the name such as `PMULL2`.

Reply:
> The default is unless otherwise specified by the instruction (such is inserts, or high operations) the unused bits of a register are always cleared to 0 on writes.

Awesome, thanks!

This is the same behavior as for x86 when looking at 128-bit vs 256-bit registers (just with 64-bit vs 128-bit on ARM). However, it differs from x86 when looking at scalar vs vector operations (x86 preserves the upper bits, through bit 128 and clears bits 128-256; where-as ARM just clears all upper bits).

Reply:
@CarolEidt, just for a sanity check. Is this register allocator aware of these parts (scalar vs vector64 vs vector128)? Is there anything we need to set to tell it when an operation will preserve vs zero the "upper bits"?

Reply:
That raises a very interesting point. The register allocator only understands fully-overwritten or RMW operands. That is, an operation that partially preserves the value of a target should have that as both a source and a target, and that source should be marked as "delayFree". It is then up to the code generator to copy it if the RMW src and dst aren't allocated the same register. I suspect there are some issues here in what we're doing today.